front/cypress/e2e/admin/analysis/survey_analysis.cy.ts

```diff
@@ -0,0 +1,143 @@
+          phaseId = phase.body.data.id;
+
+          return cy
+            .apiCreateSurveyQuestions(phaseId, [
+              'page',
+              'text',
+              'select',
+              'number',
+            ])
+            .then((survey) => {
+              const surveyFields = survey.body.data;
+
+              surveyQuestionResponses.forEach((response) => {
+                cy.apiCreateSurveyResponse({
+                  project_id: projectId,
+                  fields: {
+                    [surveyFields[1].attributes.key]: response,
+                  },
+                });
+              });
+            });
+        });
+    });
+  });
+
+  it('shows and hides summaries on the survey results page', () => {
+    cy.visit(`/admin/projects/${projectId}/phases/${phaseId}/native-survey`);
+    cy.intercept('GET', '**/insights', {
+      fixture: 'analysis_insights_survey.json',
+    });
+
+    cy.get('#e2e-analysis-summary').should('exist');
+    cy.get('#e2e-analysis-actions').click();
+    cy.get('#e2e-hide-summaries').click();
+    cy.get('#e2e-analysis-summary').should('not.exist');
+    cy.get('#e2e-show-summaries').click();
+    cy.get('#e2e-analysis-summary').should('exist');
+  });
+
+  it('adds and removes questions from the analysis', () => {
+    cy.visit(`/admin/projects/${projectId}/phases/${phaseId}/native-survey`);
+    cy.intercept('GET', '**/insights', {
+      fixture: 'analysis_insights_survey.json',
+    });
+
+    cy.get('#e2e-explore-summary').click();
+
+    // Launch modal
+    cy.get('#e2e-analysis-launch-modal').should('exist');
+    cy.get('#e2e-analysis-launch-modal-agree-button').click();
+
+    cy.get('[data-cy="e2e-analysis-summary"]').should('exist');
+    cy.get('[data-cy="e2e-analysis-question"]').should('exist');
+    cy.get('[data-cy="e2e-analysis-input-item"]').should('have.length', 12);
+    cy.get('[data-cy="e2e-analysis-custom-field-item"]').should(
+      'have.length',
+      1
+    );
+    cy.get('[data-cy="e2e-analysis-custom-field-item"]').should(
+      'contain',
+      'Question: text'
+    );
+
+    // Add question to analysis
+
+    cy.get('#e2e-analysis-toggle-show-all-questions-button').click();
+    cy.get('[data-cy="e2e-analysis-add-remove-additional-custom-field"]')
+      .first()
+      .click();
+    cy.get('#e2e-analysis-toggle-show-all-questions-button').click();
+    cy.get('[data-cy="e2e-analysis-custom-field-item"]').should(
+      'have.length',
+      2
+    );
+    cy.get('[data-cy="e2e-analysis-custom-field-item"]')
+      .first()
+      .should('contain', 'Question: text');
+    cy.get('[data-cy="e2e-analysis-custom-field-item"]')
+      .last()
+      .should('contain', 'Question: select');
+
+    // Remove question from analysis
+
+    cy.get(
+      '[data-cy="e2e-analysis-add-remove-additional-custom-field"]'
+    ).click();
+    cy.get('[data-cy="e2e-analysis-custom-field-item"]').should(
+      'have.length',
+      1
+    );
+  });
+
+  // after(() => {
```

Comment:
This is commented out for the moment because we have a bug with project deletion causing the test to fail. We are working on fixing it and un-commenting this part of the tests.